### PR TITLE
Adding Slicing Events

### PR DESCRIPTION
**Pull request by joserochh**

### CODE REVIEW

File: Source/USemLog/USemLog.Build.cs [hunk 1]
```diff
@@ -45,11 +45,12 @@ public USemLog(ReadOnlyTargetRules Target) : base(Target)
 				"UTags",
 				"UIds",
 				"UConversions",
+				"UMCGrasp",			// SL_WITH_MC_GRASP
+				"libmongo",			// SL_WITH_LIBMONGO
+                                "SlicingLogic",                 // SL_WITH_SLICING
```
Comment (andreihaidu) [inline Source/USemLog/USemLog.Build.cs]: by default this should be commented out

File: USemLog.uplugin [hunk 2]
```diff
@@ -52,6 +52,10 @@
 		{
 			"Name": "libmongo",
 			"Enabled": true
+		},
```
Comment (andreihaidu) [inline USemLog.uplugin]: here as well, by default the plugin should not depend on this

File: Source/USemLogEd/Private/SLEdModeToolkit.cpp [hunk 3]
```diff
@@ -229,7 +229,7 @@ FReply FSLEdModeToolkit::GenerateNewSemanticIds()
 		int32 TagIndex = FTags::GetTagTypeIndex(*ActItr, "SemLog");
 		if (TagIndex != INDEX_NONE)
 		{
-			FTags::AddKeyValuePair(ActItr->Tags[TagIndex], "Id", FIds::NewGuidInBase64Url(), true, *ActItr);
+			FTags::AddKeyValuePair(*ActItr, ActItr->Tags[TagIndex].ToString(), "Id", FIds::NewGuidInBase64Url(), true);
```
Comment (andreihaidu) [inline Source/USemLogEd/Private/SLEdModeToolkit.cpp]: @joserochh this seems a bit unrelated, do you know why were the changed necessary?

File: USemLog.uplugin (hunk 2 above)
Comment (joserochh) [reply USemLog.uplugin]: I'll be commenting this out and the other one.

File: Source/USemLogEd/Private/SLEdModeToolkit.cpp (hunk 3 above)
Comment (joserochh) [reply Source/USemLogEd/Private/SLEdModeToolkit.cpp]: I had a compilation Error, no function prototype matched this list of arguments so I had to modify  it. Anyway I may keep this local.

File: USemLog.uplugin (hunk 2 above)
Comment (joserochh) [reply USemLog.uplugin]: Should I disable UUtils and UMC also? @andreihaidu

Comment (andreihaidu) [reply USemLog.uplugin]: @joserochh no, those can stay since they are in the "robcog ecosystem" once we move the slicing one as well from the student project we can also leave that by default.

**Comment by joserochh** [comment]
Hi Andrei, I added some commits that takes into account your comments. You can check again and let me know if something else is needed.
